src/net.cpp

```diff
@@ -2564,6 +2564,14 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
         bool fFeeler = false;
         std::optional<Network> preferred_net;
 
+        std::vector<Network> networks;
+        for (int i = 0; i != NET_MAX; i++) {
+            Network net = static_cast<Network>(i);
+            if (g_reachable_nets.Contains(net)) {
+                networks.push_back(net);
+            }
+        }
```

Comment:
If you extend the `ReachableNets` class like this:

```diff
--- i/src/netbase.h
+++ w/src/netbase.h
@@ -102,12 +102,18 @@ public:
     [[nodiscard]] bool Contains(const CNetAddr& addr) const EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
     {   
         AssertLockNotHeld(m_mutex);
         return Contains(addr.GetNetwork());
     }

+    [[nodiscard]] std::unordered_set<Network> All() const EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+    {
+        LOCK(m_mutex);
+        return m_reachable;
+    }
+
 private:
     mutable Mutex m_mutex;

     std::unordered_set<Network> m_reachable GUARDED_BY(m_mutex){
```

Then you can pass directly `g_reachable_nets.All()` to `Select()` without an intermediate variable.

Reply:
Nice idea! I'll address it.